qcodes/instrument_drivers/Keysight/keysight_34980a.py

```diff
@@ -0,0 +1,186 @@
+import logging
+import warnings
+from functools import wraps
+from .keysight_34980a_submodules import KEYSIGHT_MODELS
+from qcodes import VisaInstrument
+from typing import List, Callable, Optional
+
+LOGGER = logging.getLogger()
+
+
+def post_execution_status_poll(func: Callable) -> Callable:
+    """
+    Generates a decorator that clears the instrument's status registers
+    before executing the actual call and reads the status register after the
+    function call to determine whether an error occurs.
+
+    Args:
+        func: function to wrap
+    """
+    @wraps(func)
+    def wrapper(self, *args, **kwargs):
+        self.clear_status()
+        retval = func(self, *args, **kwargs)
+
+        stb = self.get_status()
+        if stb:
+            warnings.warn(f"Instrument status byte indicates an error occurred "
+                          f"(value of STB was: {stb})! Use `get_error` method "
+                          f"to poll error message.",
+                          stacklevel=2)
+        return retval
+
+    return wrapper
+
+
+class Keysight34980A(VisaInstrument):
+    """
+    QCodes driver for 34980A switch/measure unit
+    """
+    def __init__(self, name, address, **kwargs):
+        """
+        Create an instance of the instrument.
+
+        Args:
+            name (str): Name used by QCoDeS. Appears in the DataSet
+            address (str): Visa-resolvable instrument address.
+        """
+        super().__init__(name, address, terminator='\n', **kwargs)
+
+        self.add_parameter(name='get_status',
+                           get_cmd='*ESR?',
+                           get_parser=int,
+                           docstring='Queries status register.')
+
+        self.add_parameter(name='get_error',
+                           get_cmd=':SYST:ERR?',
+                           docstring='Queries error queue')
+
+        self._system_slots_info_dict: Optional[dict] = None
+        self.module = dict.fromkeys(self.system_slots_info.keys())
+        self.scan_slots()
+        self.connect_message()
+
+    def scan_slots(self) -> None:
+        """
+        Scan the occupied slots and make an object for each switch matrix
+        module installed
+        """
+        for slot in self.system_slots_info.keys():
+            module_info = self.system_slots_info[slot]['module']
+            for module in KEYSIGHT_MODELS:
+                if module in module_info:
+                    name = 'slot' + str(slot)
+                    sub_mod = KEYSIGHT_MODELS[module](self, name, slot)
+                    self.module[slot] = sub_mod
+                    break
```

Comment:
The `break` is to save some time, because once a module is found, there is no need to continue the for loop (`for module in KEYSIGHT_MODELS:`), and can move to next occupied slot.
- the keys in the `KEYSIGHT_MODELS` are unique
- `module_info` will be something like "34934A-8x64", so only one module in the `module_info`

No sure if possible that `len(models_of_module_in_this_slot)` could be more than 1